### ISSUE TITLE
Remote SSH connections don't work

### ISSUE THREAD

**Issue by HarrisonWilde** [VSCodium/vscodium#460]
**Describe the bug**
I have installed the Remote Development extensions from the marketplace, in VSCodium and VS Code. In VS Code I have no problem picking a host from the list generated using my SSH config file and then entering a password (if required) to start working on remote directories. In VSCodium a password prompt will still appear and accept the input but then I get errors stating "Could not fetch remote environment" and "Failed to connect to the remote extension host server (Error: Connection error: Unauthorized client refused.)". Same thing happens with hosts that I have key-pairs etc set up for.

**Please confirm that this problem is VSCodium-specific**
- [X] This bug doesn't happen if I use Microsoft's Visual Studio Code. It only happens in VSCodium.

**Please confirm that the issue/resolution isn't already documented**
- [X] I checked [the Docs page](https://github.com/VSCodium/vscodium/blob/master/DOCS.md) and my issue is not mentioned there.

**To Reproduce**
1. Install the "Remote - SSH" extension
2. Choose a host to connect to
3. Enter password if prompted

**Desktop (please complete the following information):**
 - OS: Mac OS Catalina
 - Architecture: x64
 - Version: 1.47.2

**Comment by choosegit** [comment]
在windows x64中这个问题也出现了

**Comment by voteblake** [comment]
Having this issue with a local host OS of Ubuntu and remote host running Amazon Linux 2, both x86_64. Was thinking I was running in to https://github.com/microsoft/vscode-remote-release/issues/103, but after confirming library versions exceed the required minimum for AL2 it looks like this is VSCodium specific.

**Comment by voteblake** [comment]
Stack trace from Log (Window) output is below. There are logs in the Remote - SSH output channel as well, but nothing there indicates an error. It is able to connect and install the server side tools.

```
[2020-07-20 17:32:29.345] [renderer1] [error] [remote-connection][Management   ][aaac0…][initial][127.0.0.1:41071] received error control message when negotiating connection. Error:
[2020-07-20 17:32:29.346] [renderer1] [error] Error: Connection error: Unauthorized client refused.
    at x (file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:1868:697)
    at file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:1861:177
    at c.fire (file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:95:801)
    at v.fire (file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:430:733)
    at t.PersistentProtocol._receiveMessage (file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:435:86)
    at file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:432:243
    at c.fire (file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:95:801)
    at g.acceptChunk (file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:428:234)
    at file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:427:604
    at Socket.t (file:///usr/share/codium/resources/app/out/vs/workbench/workbench.desktop.main.js:436:655)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:166:17)
```

**Comment by HarrisonWilde** [comment]
Any update on this?

**Comment by femto-code** [comment]
This issue might be related to the fact that the "Remote - SSH" extension is strictly ruled to only work in the "official" VS Code builds by Microsoft. There are a few other extensions that won't be functional in VS Codium.

**Comment by voteblake** [comment]
That would definitely explain it @femto-code - how did you see that this was the case?

**Comment by femto-code** [comment]
> That would definitely explain it @femto-code - how did you see that this was the case?

Ah, recently I came across another issue that deals with the complete Remote Development package that seems to be blocked on VS Codium.
I attach you the links, it is quite interesting though it is worth to question why you just cannot use them.
See the last answers on #196 vor the complete #240 thread.
I hope I could help you @voteblake !

**Comment by voteblake** [comment]
Thanks @femto-code - it looks like this is a duplicate issue, the error message has changed but the root cause is the same. Appreciate you providing the links.

**Comment by gjgd** [comment]
Same issue, same error message. Remote-ssh works on VSCode for me but not on VSCodium (with exactly the same config).

**Comment by femto-code** [comment]
Thanks at @gjgd for your additional report in this issue. It supports the phenomenon described in #196 and #240.
Indeed, I support the proposal of the other devs in #240 to keep up with this problem and to find a work-around solution for this reluctant behaviour by Microsoft.
I think it is obvious why they cut off certain functionalities (like the remote tools) in great alternative build projects like VS Codium (without their telemetry in it) - that disqualifies Microsoft as a privacy-unfriendly software distributor...

**Comment by juanantoniomosquera** [comment]
same issue...

**Comment by johnebgood** [comment]
Spent 4+ hours trying to get Remote - SSH to work on VSCodium with multiple server instances and then found this still open issue. Is this going to be fixed? Can't this be fixed? If Remote - SSH won't work with VSCodium can this be closed and clearly stated somewhere? How about make VSCodium so it doesn't show extensions that won't work or shows a notification saying it won't work.

**Comment by daiyam** [comment]
@johnebgood This won't be fixed since the Remote extensions are proprietary/close-source  of Microsoft. An alternative has to be developed...

**Comment by tidux** [comment]
Can vscodium just implement a list of extension names that it lies about versions to, or is the extension actively checking for some unimplemented proprietary code path?

**Comment by daiyam** [comment]
I've add the wiki page: https://github.com/VSCodium/vscodium/wiki/Extensions-Compatibility

**Comment by khanzf** [comment]
This is a blocker for me using VS Codium. There should be an alternative.

**Comment by femto-code** [comment]
> This is a blocker for me using VS Codium. There should be an alternative.

Just a question of interest: What keeps you from using connection methods provided by your environment/file manager?

**Comment by khanzf** [comment]
I do not know about this method. Can you show me what you are referring to?
To be clear, this is for a remote machine, correct?

**Comment by raj-magesh** [comment]
@femto-code I could mount my remote filesystem using SSHFS and edit remote files using VSCodium, but would it be possible to use a remote Python environment and offload the actual execution/interactive debugging of a Python module to the remote server within VSCodium?

**Comment by femto-code** [comment]
@khanzf Yes, correct, I mean a remote machine. In KDE, you can connect seamlessly with dolphin file manager and then manage the files as if they would be on your local machine. Which OS are you using / are we talking about?

**Comment by femto-code** [comment]
@raj-magesh You're right, using SSHFS would be no problem. Concerning Python debugging, I am not sure whether this coukld be accomplished externally except for using the terminal within VSCodium, would you be able to get the debugging stuff done this way?

**Comment by khanzf** [comment]
@femto-code It would be a problem if your situation isn't really simple, but in my case there are permissions issues, a locally running database and even different OS's (Linux vs FreeBSD).

ssh-fs is great, but does not really meet the use-case here.
